src/analytics/SRGAnalytics.js

```diff
@@ -0,0 +1,873 @@
+   *
+   * @returns {Object} JSON to be sent to TagCommander
+   */
+  getEventLabels(eventName) {
+    const labels = {
+      event_id: eventName,
+      event_timestamp: SRGAnalytics.now(),
+      media_dvr_window_length: 0,
+      media_dvr_window_offset: 0,
+      media_is_dvr: false,
+      media_is_live: false,
+      media_mute: this.player.muted() ? '1' : '0',
+      media_playback_rate: this.player.playbackRate(),
+      media_position: this.currentTime(),
+      media_quality: this.srcMediaData.mediaData.quality,
+      // TODO use media_is_dvr, media_is_live to define peach media_stream_type
+      media_subtitles_on: this.isTextTrackEnabled(),
+      media_volume: (this.player.volume() * 100).toFixed(0),
+      navigation_environment: this.environment,
+    };
+
+    if (this.isAudioTrackEnabled()) {
+      labels.media_audio_track = this.getCurrentAudioTrack();
+    }
+
+    if (this.isTextTrackEnabled()) {
+      labels.media_subtitle_selection = this.getCurrentTextTrack();
+    }
+
+    // DVR or Live related labels
+    if (!this.isMediaOnDemand()) {
+      labels.media_is_live = true;
+      labels.media_position = this.getElapsedPlaybackTime();
+    }
+
+    // DVR related labels
+    if (this.isMediaDvr()) {
+      labels.media_dvr_window_offset = this.getDvrWindowPosition() | 0;
+      labels.media_dvr_window_length = this.getDvrWindowSize() | 0;
+
+      labels.media_is_dvr = true;
+
+      labels.media_timeshift = [PlayerEvents.PLAY, PlayerEvents.PAUSE].includes(
+        eventName
+      )
+        ? this.timeShifted()
+        : 0;
+    }
+
+    return labels;
+  }
+
+  /**
+   * Set all internal labels to be sent to TagCommander. Internal labels are assigned once at initialisation time.
+   */
+  getInternalLabels() {
+    const data = {
+      media_bu_distributer: this.srcMediaData.mediaData.vendor,
+      media_chromecast_selected: Boolean(this.player.tech(true).isCasting),
+      media_embedding_url: document.referrer,
+      media_player_display: 'default', // TODO implement if it still relevant
+      media_player_name: 'pillarbox-web', // TODO add a property playerName in the constructor with a default value ?
+      media_player_version: this.playerVersion,
+      media_url: this.srcMediaData.src,
+    };
+    const analyticsMetadata =
+      this.srcMediaData.mediaData.analyticsMetadata || {};
+
+    window.tc_vars = Object.assign({}, window.tc_vars, data, analyticsMetadata);
+  }
+
+  /**
+   * Check if the audio track is enabled.
+   *
+   * @returns {Boolean} __true__ if enabled __false__ otherwise.
+   */
+  isAudioTrackEnabled() {
+    return !!this.getCurrentAudioTrack();
+  }
+
+  /**
+   * Check if the text track is enabled.
+   *
+   * @returns {Boolean} __true__ if enabled __false__ otherwise.
+   */
+  isTextTrackEnabled() {
+    return !!this.getCurrentTextTrack();
+  }
+
+  /**
+   * Heart beat, current position of a AoD/VoD (every 30s)
```

Comment:
```suggestion
   * Heart beat, current position of a AoD/VoD (every 30s).
```

Reply:
I'll also update the reference link to the documentation